app/controllers/scholars_archive/handles_controller.rb

```diff
@@ -0,0 +1,97 @@
+module ScholarsArchive
+  class HandlesController < ApplicationController
+    before_filter :verify_handle_id, only: [:handle_show, :handle_download]
+    skip_before_action :check_d2h_http_header_auth
+
+    def handle_file_404
+      @handle_uri = "#{params[:handle_id]}/#{params[:id]}"
+      @file = "#{params[:file]}"
+      work = find_work
+      @work_path = "/concern/#{work.class.to_s.downcase.pluralize}/#{work.id}"
+    end
+
+    def handle_work_404
+
+    end
+
+    def handle_show
+      work = find_work
+      if work.nil?
+        ScholarsArchive::HandleErrorLoggingService.log_no_work_found_error(params)
+        redirect_to "/404/work_not_found"
```

Comment:
Some of the options in there may allow you to use more of the built-in Rails stuff and not need custom 404 routes and views just to display the extra information.

Reply:
In order to have multiple different 404 pages with useful information, we definitely need to make some custom routes for the 404. Rails only knows about the /404 route, so no matter what we would need to make at least one new route. I figured we could leave the original 404 page as needed if someone goes anywhere else in the app that returns a 404. Otherwise, when they get the 404 messages from the handle redirect, they get some useful info and can make a decision after that.

I can change it so it renders a page with the proper status code for sure though.